test/test-listeners.js

```diff
@@ -119,6 +136,19 @@ it('test manual lifecycle', () => {
   assert(count === 2, 'listener was removed');
 });
 
+it('test synchronous event handling', () => {
```

Comment:
Yes indeed, this test does fail without the related change in `x-element.js`. Writing this test made it additionally-clear that this is indeed something worth fixing, by the way.